modules/programs/aerc-accounts.nix

```diff
@@ -208,7 +213,12 @@ in {
 
       sourceCfg = account:
         if account.mbsync.enable || account.offlineimap.enable then
-          mkConfig.maildir account
+          if account.mbsync.enable && account.mbsync.flatten == null
+          && account.mbsync.subFolders == "Maildir++" then
+            mkConfig.maildirpp account
+          else
+            mkConfig.maildir account
```

Comment:
I would probably rewrite this part like so:

```
if account.mbsync.enable && account.mbsync.flatten == null && account.mbsync.subFolders == "Maildir++" then
    mkConfig.maildirpp account
else if account.mbsync.enable || account.offlineimap.enable then
    mkConfig.maildir account
```

Reply:
Done!